lib/matplotlib/figure.py

```diff
@@ -1528,107 +956,868 @@ def subplots(self, nrows=1, ncols=1, sharex=False, sharey=False,
+        x, y : float
+            The position to place the text. By default, this is in figure
+            coordinates, floats in [0, 1]. The coordinate system can be changed
+            using the *transform* keyword.
 
-            """
-            r0, *rest = inp
-            for j, r in enumerate(rest, start=1):
-                if isinstance(r, str):
-                    raise ValueError('List layout specification must be 2D')
-                if len(r0) != len(r):
-                    raise ValueError(
-                        "All of the rows must be the same length, however "
-                        f"the first row ({r0!r}) has length {len(r0)} "
+        s : str
+            The text string.
+
+        fontdict : dict, optional
+            A dictionary to override the default text properties. If not given,
+            the defaults are determined by :rc:`font.*`. Properties passed as
+            *kwargs* override the corresponding ones given in *fontdict*.
+
+        Returns
+        -------
+        `~.text.Text`
+
+        Other Parameters
+        ----------------
+        **kwargs : `~matplotlib.text.Text` properties
+            Other miscellaneous text parameters.
+
+            %(Text)s
+
+        See Also
+        --------
+        .Axes.text
+        .pyplot.text
+        """
+        effective_kwargs = {
+            'transform': self.transSubfigure,
+            **(fontdict if fontdict is not None else {}),
+            **kwargs,
+        }
+        text = Text(x=x, y=y, text=s, **effective_kwargs)
+        text.set_figure(self)
+        text.stale_callback = _stale_figure_callback
+
+        self.texts.append(text)
+        text._remove_method = self.texts.remove
+        self.stale = True
+        return text
+
+    @docstring.dedent_interpd
+    def colorbar(self, mappable, cax=None, ax=None, use_gridspec=True, **kw):
+        """%(colorbar_doc)s"""
+        if ax is None:
+            ax = self.gca()
+            if (hasattr(mappable, "axes") and ax is not mappable.axes
+                    and cax is None):
+                cbook.warn_deprecated(
+                    "3.4", message="Starting from Matplotlib 3.6, colorbar() "
+                    "will steal space from the mappable's axes, rather than "
+                    "from the current axes, to place the colorbar.  To "
+                    "silence this warning, explicitly pass the 'ax' argument "
+                    "to colorbar().")
+
+        # Store the value of gca so that we can set it back later on.
+        current_ax = self.gca()
+        artists = self._get_draw_artists(self.canvas.get_renderer())
+        if cax is None:
+            if (use_gridspec and isinstance(ax, SubplotBase)
+                    and not self.get_constrained_layout()):
+                cax, kw = cbar.make_axes_gridspec(ax, **kw)
+            else:
+                cax, kw = cbar.make_axes(ax, **kw)
+        artists = self._get_draw_artists(self.canvas.get_renderer())
```

Comment:
Why is `_get_draw_artists` called twice here? And then `artists` is not used for anything.

Reply:
both calls removed....  Not sure either.